src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/FailureCauseBuildAction.java

```diff
@@ -342,7 +343,17 @@ private static void adjustProjectDisplayName(
     private static Set<Run<?, ?>> getDownstreamBuilds(
             final Run build) {
 
-        Set<Run<?, ?>> foundDbf = new TreeSet<Run<?, ?>>();
+        Set<Run<?, ?>> foundDbf = new TreeSet<Run<?, ?>>(new Comparator<Run<?, ?>>() {
+            @Override
+            public int compare(Run<?, ?> o1, Run<?, ?> o2) {
+                final int res = o1.getParent().getFullName().compareTo(o2.getParent().getFullName());
```

Comment:
> Yes but does that matter?

If doesn't matter for you - it doesn't matter for me as well. I assumed that initial idea of having this `TreeSet` was to have sorted list of jobs with permanent order in UI. If it's not a case, I can replace it by `HashSet` - it doesn't care about `compareTo` at all and it will fix this bug as well.

> Checking for null would be simpler as you would only need to check if o1 and o2 are null before proceeding.

But I don't need to check it in comparator.
There is already this check in place (just before pushing to `TreeSet`). Line 364.

I need this check only it's possible that `run` is not `null`, but `parent` of `run` is `null`. I didn't except that it's possible.